src/internet_identity/src/storage/storable_persistent_state.rs

```diff
@@ -0,0 +1,126 @@
+            max_inflight_captchas: DEFAULT_MAX_INFLIGHT_CAPTCHAS,
+        }
+    }
+}
+
+impl From<PersistentState> for StorablePersistentState {
+    fn from(persistent_state: PersistentState) -> Self {
+        let time = time();
+        Self {
+            archive_state: persistent_state.archive_state,
+            canister_creation_cycles_cost: persistent_state.canister_creation_cycles_cost,
+            registration_rate_limit: persistent_state
+                .registration_rate_limit
+                .unwrap_or(DEFAULT_RATE_LIMIT_CONFIG),
+            active_anchor_stats: persistent_state
+                .active_anchor_stats
+                .unwrap_or(ActivityStats::new(time)),
+            domain_active_anchor_stats: persistent_state
+                .domain_active_anchor_stats
+                .unwrap_or(ActivityStats::new(time)),
+            active_authn_method_stats: persistent_state
+                .active_authn_method_stats
+                .unwrap_or(ActivityStats::new(time)),
+            latest_delegation_origins: persistent_state
+                .latest_delegation_origins
+                .unwrap_or_default(),
+            max_num_latest_delegation_origins: persistent_state
+                .max_num_latest_delegation_origins
+                .unwrap_or(DEFAULT_MAX_NUM_DELEGATION_ORIGINS),
+            max_inflight_captchas: persistent_state
+                .max_inflight_captchas
+                .unwrap_or(DEFAULT_MAX_INFLIGHT_CAPTCHAS),
+        }
+    }
+}
+
+#[cfg(not(test))]
+fn time() -> Timestamp {
+    ic_cdk::api::time()
+}
+
+/// This is required because [ic_cdk::api::time()] traps when executed in a non-canister environment.
+#[cfg(test)]
+fn time() -> Timestamp {
+    std::time::SystemTime::now()
+        .duration_since(std::time::SystemTime::UNIX_EPOCH)
+        .unwrap()
+        .as_nanos() as u64
+}
+
+impl From<StorablePersistentState> for PersistentState {
+    fn from(storable_persistent_state: StorablePersistentState) -> Self {
+        Self {
+            archive_state: storable_persistent_state.archive_state,
+            canister_creation_cycles_cost: storable_persistent_state.canister_creation_cycles_cost,
+            registration_rate_limit: Some(storable_persistent_state.registration_rate_limit),
+            active_anchor_stats: Some(storable_persistent_state.active_anchor_stats),
+            domain_active_anchor_stats: Some(storable_persistent_state.domain_active_anchor_stats),
+            active_authn_method_stats: Some(storable_persistent_state.active_authn_method_stats),
+            latest_delegation_origins: Some(storable_persistent_state.latest_delegation_origins),
+            max_num_latest_delegation_origins: Some(
+                storable_persistent_state.max_num_latest_delegation_origins,
+            ),
+            max_inflight_captchas: Some(storable_persistent_state.max_inflight_captchas),
+        }
+    }
+}
```

Comment:
maybe add some basic test(s)? (e.g. a round-trip test for for the `From`-impls, and a stability test for the `Default`-impl?)

Reply:
Added tests for both. Good point.